src/python/pants/engine/internals/platform_rules.py

```diff
@@ -39,7 +39,9 @@ def current_platform(
         if env_tgt.val.has_field(DockerPlatformField):
             return Platform(env_tgt.val[DockerPlatformField].normalized_value)
         if env_tgt.val.has_field(RemotePlatformField):
-            return Platform(env_tgt.val[RemotePlatformField].value)
+            remote_platform = env_tgt.val[RemotePlatformField].value
```

Comment:
I am very confused:

 * Why mypy was satisfied with this before since I *think* `Enum` behaves the same way.
  * Why it neede dso much extra help with the `remote_platform` var callout instead of just a check on `.value`.
  
  This is to resolve:
  
  ```
  Partition #2 - python-default, ['CPython==3.11.*']:
src/python/pants/engine/internals/platform_rules.py: note: In function "current_platform":
src/python/pants/engine/internals/platform_rules.py:42:29: error: Argument 1 to "Platform" has incompatible type "str | None"; expected "str"  [arg-type]
                return Platform(env_tgt.val[RemotePlatformField].value)
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
I don't understand it either, but I have seen mypy get confused about this sort of thing elsewhere at times.